### PR TITLE
MixedCaseNameCheck Enhancements (without AutoFix)

### PR DESCRIPTION
**Pull request by vladlemberg**
### Description:

Please refer: https://github.com/osmlab/atlas-checks/issues/393

Check enhancements: https://github.com/osmlab/atlas-checks/issues/540

### Potential Impact:

List potential impact to downstream libraries here (ex. atlas-checks, atlas-generator).

### Unit Test Approach:

Add unit test for bug fix and more corner case coverage  

### Test Results:

passed

### CODE REVIEW

**Review by atiannicelli — DISMISSED** [review]

**Review by Bentleysb — APPROVED** [review]

File: src/main/java/org/openstreetmap/atlas/checks/validation/tag/MixedCaseNameCheck.java [hunk 1]
```diff
@@ -127,6 +135,12 @@ public boolean validCheckForObject(final AtlasObject object)
                                 .contains(object.tag(ISOCountryTag.KEY).toUpperCase())
                         // And have a name tag
                         && Validators.hasValuesFor(object, NameTag.class)
+                        // Excluding names with following tags to reduce number of legit
+                        // MixedCaseNames
+                        && !Validators.hasValuesFor(object, BrandTag.class)
+                        && !Validators.hasValuesFor(object, ShopTag.class)
+                        && !Validators.hasValuesFor(object, AmenityTag.class)
+                        && !Validators.hasValuesFor(object, LeisureTag.class)
```
Comment (sayas01) [inline src/main/java/org/openstreetmap/atlas/checks/validation/tag/MixedCaseNameCheck.java:143]: this could be combined to `&& !Validators.hasValuesFor(object, BrandTag.class, ShopTag.class, AmenityTag.class, LeisureTag.class)`

**Review by sayas01 — APPROVED** [review]
LGTM! 👍

**Review by Bentleysb — APPROVED** [review]
